Nil Pointer in watchers while running in Arbitrum

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x58 pc=0x7ff75be2a3d5]

goroutine 32836 [running]:
github.com/livepeer/go-livepeer/eth/watchers.(*SenderWatcher).handleBlockEvents(0xc002c89fd0, {0xc000882100, 0x1c, 0x0})
        D:/a/go-livepeer/go-livepeer/eth/watchers/senderwatcher.go:152 +0x55
created by github.com/livepeer/go-livepeer/eth/watchers.(*SenderWatcher).Watch
        D:/a/go-livepeer/go-livepeer/eth/watchers/senderwatcher.go:116 +0x385
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x58 pc=0x7ff75be2c595]

goroutine 33 [running]:
github.com/livepeer/go-livepeer/eth/watchers.(*ServiceRegistryWatcher).handleBlockEvents(0xc000283f80, {0xc000882100, 0x1c, 0x0})
        D:/a/go-livepeer/go-livepeer/eth/watchers/serviceRegistryWatcher.go:61 +0x55
github.com/livepeer/go-livepeer/eth/watchers.(*ServiceRegistryWatcher).Watch(0xc000211e80)
        D:/a/go-livepeer/go-livepeer/eth/watchers/serviceRegistryWatcher.go:49 +0x20a
created by main.main
        D:/a/go-livepeer/go-livepeer/cmd/livepeer/livepeer.go:587 +0x43f8
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x58 pc=0x7ff75be2f355]

goroutine 32748 [running]:
github.com/livepeer/go-livepeer/eth/watchers.(*UnbondingWatcher).handleBlockEvents(0xc002c8ffd0, {0xc000882100, 0x1c, 0x0})
        D:/a/go-livepeer/go-livepeer/eth/watchers/unbondingwatcher.go:74 +0x55
created by github.com/livepeer/go-livepeer/eth/watchers.(*UnbondingWatcher).Watch
        D:/a/go-livepeer/go-livepeer/eth/watchers/unbondingwatcher.go:62 +0x2af
```

--- a/CHANGELOG_PENDING.md
+++ b/CHANGELOG_PENDING.md
@@ -15,6 +15,7 @@
 #### Transcoder
 
 ### Bug Fixes 🐞
+- \#2267 Fix nil pointer in the block header logs (@leszko)
 
 #### General
 

--- a/eth/blockwatch/block_watcher.go
+++ b/eth/blockwatch/block_watcher.go
@@ -637,7 +637,12 @@ func (w *Watcher) enrichWithL1BlockNumber(header *MiniHeader) (*MiniHeader, erro
 	if header == nil || header.L1BlockNumber != nil {
 		return header, nil
 	}
-	return w.client.HeaderByHash(header.Hash)
+	fetchedBlock, err := w.client.HeaderByHash(header.Hash)
+	if err != nil {
+		return header, err
+	}
+	header.L1BlockNumber = fetchedBlock.L1BlockNumber
+	return header, nil
 }
 
 func isUnknownBlockErr(err error) bool {

--- a/eth/blockwatch/block_watcher_test.go
+++ b/eth/blockwatch/block_watcher_test.go
@@ -527,6 +527,7 @@ func TestEnrichWithL1_One(t *testing.T) {
 	header := &MiniHeader{
 		Number: big.NewInt(FakeBlockNumber),
 		Hash:   common.HexToHash(FakeHash),
+		Logs:   []types.Log{logStub},
 	}
 	events := []*Event{{BlockHeader: header}}
 
@@ -536,6 +537,7 @@ func TestEnrichWithL1_One(t *testing.T) {
 	assert.Equal(common.HexToHash(FakeHash), res[0].BlockHeader.Hash)
 	assert.Equal(big.NewInt(FakeBlockNumber), res[0].BlockHeader.Number)
 	assert.Equal(big.NewInt(FakeL1BlockNumber), res[0].BlockHeader.L1BlockNumber)
+	assert.Contains(res[0].BlockHeader.Logs, logStub)
 }
 
 func TestEnrichWithL1_Multiple(t *testing.T) {

